fedora_messaging/exceptions.py

```diff
@@ -75,12 +75,25 @@ def __str__(self):
 
 
 class PublishException(BaseException):
-    """Base class for exceptions related to publishing."""
+    """
+    Base class for exceptions related to publishing.
+
+    Args:
+        reason (str): The reason the server gave for the publish error.
+    """
 
     def __init__(self, reason=None, **kwargs):
         super(PublishException, self).__init__(**kwargs)
         self.reason = reason
 
+    def __str__(self):
+        if self.reason:
+            return "Publish error: " + self.reason
```

Comment:
It'd be good to call str on self.reason so users can pass in non-text objects (for example, a lower-level exception)